src/network/networkd-util.c

```diff
@@ -119,6 +117,49 @@ int config_parse_address_family_with_kernel(
         return 0;
 }
 
+int config_parse_ip_masquerade(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        AddressFamily a, *ret = data;
+        int r;
+
+        if (isempty(rvalue)) {
+                *ret = ADDRESS_FAMILY_NO;
+                return 0;
+        }
+
+        r = parse_boolean(rvalue);
+        if (r >= 0) {
+                if (r)
+                        log_syntax(unit, LOG_WARNING, filename, line, 0,
+                                   "IPMasquerade=%s is deprecated, and it is handled as \"both\" instead of \"ipv4\". "
+                                   "Please use \"both\" or \"ipv4\".",
+                                   rvalue);
+
+                *ret = r ? ADDRESS_FAMILY_YES : ADDRESS_FAMILY_NO;
```

Comment:
Shouldn't this be ADDRESS_FAMILY_IPV4 instead, i.e. warn, but keep semantics unchanged?

Reply:
But you also commented "I know I'm late to the party, but I'm not convinced the decision to make "yes"=="ipv4" good long term. I understand backwards compat is important, but I'd expect most people want to do both if they want at least one."
Now I am confused...

Reply:
I mean, if we choose `yes` == `ipv4` now, it is hard to change that to `yes` == `both` in the future.

Reply:
Sorry for not being clearer. The previous version of the patch made `yes` == `ipv4` silently, and I meant that we should keep that (for backwards compatibility), but warn, so that people stop using it.

Reply:
OK. Thank you for the clarification. Let's keep the backward compatibility.